framework/src/utils/MooseUtils.C

```diff
@@ -858,12 +858,24 @@ fileSize(const std::string & filename)
 }
 
 std::string
-realpath(const std::string & path)
+mooseRealpath(const std::string & path)
 {
 #ifdef LIBMESH_HAVE_PETSC
   char dummy[PETSC_MAX_PATH_LEN];
+#if defined(PETSC_HAVE_REALPATH)
+  // If "realpath" is adopted by PETSc and
+  // "path" does not exist, then PETSc will print a misleading message.
+  // [0]PETSC ERROR: Error in external library
+  // [0]PETSC ERROR: realpath()
+  // [0]PETSC ERROR: See https://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
+  // [0]PETSC ERROR: Petsc Release Version 3.13.3, unknown
+  // We here override the misleading message with a better one.
+  if (!realpath(path.c_str(), dummy))
```

Comment:
I'm guessing you changed the name so that there's not confusion here about where realpath is coming from? If you want it to be clear that it is not coming from the `MooseUtils` namespace, then you could do `::realpath` here to denote that this is coming from the global namespace.

Reply:
Right.  `realpath` is a system function. Sometimes, compiler complains  about this. Let me change it to `::realpath`